src/cpp/middleware/fastdds/FastDDSMiddleware.cpp

```diff
@@ -1242,5 +1260,15 @@ bool FastDDSMiddleware::matched_replier_from_bin(
     return rv;
 }
 
+int16_t FastDDSMiddleware::get_domain_id_from_env(){
+    int16_t agent_domain_id = 0;
+    const char * agent_domain_id_env = std::getenv( "ROS_DOMAIN_ID" );
```

Comment:
IMO The override of Micro XRCE-DDS Agent shall be ROS 2 agnostic. What about using `XRCE_DOMAIN_ID_OVERRIDE`?

CC: @Acuadros95

Reply:
Then the compose would look like this if you have defined `ROS_DOMAIN_ID` before.
```yaml
microros:
  image: husarion/micro-ros-agent:humble-overrice-domain-id
  network_mode: host
  ipc: host
  devices:
    - ${SERIAL_PORT:?err}
  environment:
    - SERIAL_PORT
    - RMW_IMPLEMENTATION=rmw_fastrtps_cpp
    - XRCE_DOMAIN_ID_OVERRIDE=${ROS_DOMAIN_ID}
    # - XRCE_DOMAIN_ID_OVERRIDE=2
  command: serial -D $SERIAL_PORT serial -b 576000 # -v6
```

Not bad. I can do this. ;)

Reply:
I agree with @pablogs9, looks cleaner